pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,49 @@
+package declarativeconfig
+
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+type ClaimMapping struct {
+	PathToClaim    string `yaml:"pathToClaim,omitempty"`
+	MapToClaimName string `yaml:"mapTo,omitempty"`
+}
+
+type OIDCConfig struct {
+	Issuer                    string `yaml:"issuer,omitempty"`
+	CallbackMode              string `yaml:"mode,omitempty"`
+	ClientID                  string `yaml:"clientID,omitempty"`
+	ClientSecret              string `yaml:"clientSecret,omitempty"`
+	DisableOfflineAccessScope bool   `yaml:"disableOfflineAccessScope,omitempty"`
+	DontUseClientSecretConfig bool   `yaml:"dontUseClientSecretConfig,omitempty"`
```

Comment:
> disableOfflineAccessScope is currently used by ACSCS only (?) and dontUseClientSecretConfig for clients like the auth0 ones, right?

`dontUseClientSecretConfig` can be selected in the UI for all OIDC auth providers. `disableOfflineAccessScope` is used by ACSCS and we don't display it in the UI, but it can be used for other OIDC auth providers as well via API.

> I'd prefer the clear documentation and "everything-and-the-kitchen-sink" as configuration

Agreed. I would prefer to have a stricter config where every field is documented. I see no need to gatekeep from declarative config users any parameters which are exposed via API.

Reply:
Sounds good to me, with the additional comments in code explaining the field a bit more (especially the `disableOfflineAccessScope` one, as it's the first time we expose it any form of API).